docs/src/main/asciidoc/key-values.adoc

```diff
@@ -0,0 +1,101 @@
+== Key Values
+
+Azure App Configuration Supports multiple types of key values, some of which have special features built into them.
+
+=== Placeholders
+
+The client library supports configurations with `${}`-style environment placeholders. If referencing a Azure App Configuration key with a placeholder any prefix needs to be removed in the reference. For example: `/application/config.message` is referenced as `${config.message}`.
+
+NOTE: The prefix being removed matches the value `spring.cloud.azure.appconfiguration.stores[0].selects[0].key-filter`.
+
+=== JSON
+
+Configurations that have the a content-type `application/json` are processed as json objects. This enables one configuration mapping to a complex object in a `@Configuration`. The json key `/application/config.colors` with the value
+
+[source,json,indent=0]
+----
+{
+	"Red": {
+		"value": [255, 0, 0]
+	},
+	"Blue": {
+		"value": [0, 255, 0]
+	},
+	"Green": {
+		"value": [0, 0, 255]
+	}
+}
+----
+
+maps to
+
+[source,java,indent=0]
+----
+@ConfigurationProperties(prefix = "config")
+public class MyConfigurations {
+
+    private HashMap<String, Color> colors;
+
+}
+----
+
+=== Key Vault References
+
+Azure App Configuration and its client libraries support referencing secrets stored in Key Vault. In App Configuration keys can be created the have values that are URIs that map to secrets in a Key Vault. This enables using secrets as configuration without storing them as configurations.
+
+Your application uses the client provider to retrieve Key Vault references, just as it does for any other keys stored in App Configuration. Because the client recognizes the keys as Key Vault references, they have a unique content-type, the client will connect to Key Vault to retrieve their values.
+
+NOTE: Key Vault only allows for secrets to be retrieved one at a time, so each key vault reference stored in App Configuration will result in a pull against Key Vault.
+
+==== Creating Key Vault References
+
+You can easily create a Key Vault reference in the Azure Portal in the Configuration explorer using the Create -> Key Vault reference option. You will be able to select a secret to make a reference to from any Key Vault you have access to. You can also create arbitrary Key Vault references using the Input option.
```

Comment:
For clarity: "...  select a secret to reference, this can be from any of Key Vaults you have access to>"

what is an arbitrary reference?

Reply:
The input option allows for any value to be set. It doesn't need to be a key vault, for secret provider.

Reply:
Makes sense, could you add this to the sentence?